test/integration/kubernetes_deploy_test.rb

```diff
@@ -907,6 +907,19 @@ def test_ejson_secrets_respects_no_prune_flag
     ejson_cloud.assert_secret_present('ejson-keys', ejson: false)
   end
 
+  def test_deploy_task_doesnt_prune_ejson_keys_secret
+    assert_deploy_success(deploy_fixtures("secrets"))
+    assert_deploy_success(deploy_fixtures("hello-cloud", subset: %w(role.yml)))
+
+    refute_logs_match(%r{The following resources were pruned:.*secret( "|\/)ejson-keys})
+    assert_logs_match_all([
+      "Secret ejson-keys has annotation kubectl.kubernetes.io/last-applied-configuration.",
+      "kubernetes-deploy will remove this annotation to ensure this secret does not get pruned.",
+      "Successfully removed kubectl.kubernetes.io/last-applied-configuration from ejson-keys secret.",
+      %r{The following resources were pruned:.*secret( "|\/)my-secret},
+    ], in_order: true)
```

Comment:
this test should also actually `get` the secret and confirm the annotation is not there anymore